components/features/event/event-form-community-selector.tsx

```diff
@@ -0,0 +1,203 @@
+    value: string;
+    label: string;
+  } | null>(() => {
+    const selectedCommunity = options.find(
+      (o) => o.value === form.getValues().communityId,
+    );
+    if (selectedCommunity) {
+      return { value: selectedCommunity.value, label: selectedCommunity.label };
+    }
+    return null;
+  });
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      const input = inputRef.current;
+      if (!input) {
+        return;
+      }
+
+      if (!isOpen) {
+        setOpen(true);
+      }
+
+      if (event.key === "Escape") {
+        input.blur();
+      }
+    },
+    [isOpen],
+  );
+
+  const handleBlur = useCallback(() => {
+    setOpen(false);
+    setSearch(selected?.label ?? "");
+  }, [selected]);
+
+  const handleSelectOption = useCallback(
+    (option: { value: string; label: string }) => {
+      setSelected(option);
+      setSearch(option.label);
+      form.setValue("communityId", option.value);
+      setOpen(false);
+      form.trigger("communityId");
+      inputRef.current?.blur();
+    },
+    [form, inputRef],
+  );
+
+  useEffect(() => {
+    const selectedCommunity = options.find(
+      (o) => o.value === form.getValues().communityId,
+    );
+
+    setSearch(selectedCommunity?.label ?? "");
+  }, [options, form, selectableCommunities]);
+
+  if (selectableCommunities.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className="flex flex-col gap-4">
+      <Heading level={3}>Link event to one your community (optional)</Heading>
+
+      <FormField
+        control={form.control}
+        name="communityId"
+        render={({ field }) => {
+          return (
+            <CommandPrimitive onKeyDown={handleKeyDown} shouldFilter={false}>
+              <div className="relative">
+                <CommandInput
+                  ref={inputRef}
+                  value={search}
+                  typeof="search"
+                  onValueChange={setSearch}
+                  onBlur={handleBlur}
+                  onFocus={() => setOpen(true)}
+                  placeholder={t("community-selector-placeholder")}
+                  className="!h-12 px-4 text-base rounded bg-custom-input-bg border border-custom-input-border"
+                  containerClassName="border-none p-0"
+                />
+                {!!field.value && (
+                  <Button
+                    type="button"
+                    variant="ghost"
+                    size="icon"
+                    className="backdrop-blur-sm self-center absolute top-1/2 -translate-y-1/2 right-1"
+                    onClick={() => {
+                      setSearch("");
+                      setSelected(null);
+                      field.onChange(null);
+                      form.trigger("communityId");
+                    }}
+                  >
+                    <XIcon className="h-3 w-3" />
+                    <span className="sr-only">Clear</span>
```

Comment:
should this text be in a i18n too ?

Reply:
[d19844d](https://github.com/samouraiworld/zenao/pull/652/commits/d19844dc8799c7327f01268c5306c5c8b2208326)